pkgs/dmd/version-catalog.nix

```diff
@@ -0,0 +1,38 @@
+rec {
+  supportedVersions = {
+    source = builtins.fromJSON (
+      builtins.readFile ./supported-source-versions.json
+    );
+
+    binary = builtins.fromJSON (
+      builtins.readFile ./supported-binary-versions.json
+    );
+  };
+
+  getSourceVersion = version:
+    assert builtins.hasAttr version supportedVersions.source; let
+      componentHashes = supportedVersions.source."${version}";
+    in
+      import ./generic.nix ({
+          inherit version;
+          dmdSha256 = componentHashes.dmd;
+          phobosSha256 = componentHashes.phobos;
+          toolsSha256 = componentHashes.tools;
+        }
+        // (
+          if componentHashes ? "druntime"
+          then {
+            druntimeSha256 = componentHashes.druntime;
+          }
+          else {}
+        ));
```

Comment:
I'll try to push this weekend a refactoring commit which moves the logic for generating package versions based on the `pkg/<name>/version-catalog.nix` file from `pkgs/default.nix` (which is written in a way that it assumes it's called from a flake-parts-style flake.nix file) to `lib/` and expose a `default.nix` that you should be able to run the old-school `nix-build` on.

Reply:
Hey @dukc I've finally pushed `default.nix` compatibility bridge for the pre-flakes world which exposes every flake output available for Nix Flakes users, i.e. we will have feature-parity whether or not one uses flakes. Additionally, I've added a readme file with usage instructions.

> This looks like it could be used to build, say 2.100, from the terminal with nix-build, so there's no need for 2.xxx.nix files anymore, even if one has no flakes. Right? What's the precise invocation?

Indeed! Here it is:

```sh
nix-build --attr packages.x86_64-linux.dmd-2_100_2
```